extensions/guacamole-auth-radius/src/main/java/org/apache/guacamole/auth/radius/RadiusConnectionService.java

```diff
@@ -129,6 +133,18 @@ private RadiusAuthenticator setupRadiusAuthenticator(RadiusClient radiusClient)
         if (radAuth == null)
             throw new GuacamoleException("Could not get a valid RadiusAuthenticator for specified protocol: " + confService.getRadiusAuthProtocol());
 
+        // For MSCHAPv1/2, we need MD4 support
+        if (radAuth instanceof MSCHAPv1Authenticator
+                || radAuth instanceof MSCHAPv2Authenticator) {
+            
+            Security.addProvider(new Provider("MD4", 0.00, "MD4 for MSCHAPv1/2 RADIUS") {
```

Comment:
`setupRadiusAuthenticator()` is invoked from within `authenticate()`, so this will result in `Security.addProvider(...)` being invoked each time someone authenticates. It's unclear to me within the documentation for `addProvider()` whether this will result in multiple copies of the MD4 implementation being added.

Perhaps this should be moved to a location where it occurs only once?

Is it known that no other auth protocols use MD4?

Reply:
Yeah, probably a good point.  I could move it to the constructor for the `RadiusConnectionService` class, but not sure if that would still happen each time someone logged in, or would be a one-shot thing?

> Is it known that no other auth protocols use MD4?

I do not know one way or the other if others require it.

Reply:
Okay, I've put the initialization for this into the Guice module, which I think will suffice to load it only a single time.

I queried the JRadius repository for any other instances of "MD4", and the MSCHAP library is the only place it shows up, so I think that covers the requirement:
https://github.com/coova/jradius/search?q=MD4&unscoped_q=MD4